Implanting in all parts of the body + crayons

stonepillars opened the pull request:
<!-- The text between the arrows are comments - they will not be visible on your PR. -->
<!-- To automatically tag this PR, add the uppercase label(s) surrounded by brackets below, for example: [LABEL] -->
[feature]

## About the PR <!-- Describe the Pull Request here. What does it change? What other things could this impact? -->
### The Implants
Add the ability to implant and remove implants from individual limbs, for implants to specify if they should create paper wads when removed (i.e. bullets, crayons, etc.)

### The Crayons
Crayons can now be shoved up the noses of carbon-based life.

It borrows the force-feeding code from pills, and it takes the same amount of time and (lack of) motion that
force-feeding a pill does.

Crayons deal 15 BRAIN when inserted and have a 20% chance per life cycle of dealing 1 BRAIN while the crayon is implanted.

**For more information about each change, view the extensive commit messages. They're much longer than this short description**.

## Why's this needed? <!-- Describe why you think this should be added to the game. -->
### The Implants
It was needed to implement the crayons, and was wanted requested, specifically by @zjdtmkhzt 

### The Crayons
You can shove crayons up your own nose for a suicide, and obviously this is very effective,
but, what if you could do it against _others_? It's a fun, simple gimmicky way to give people
BRAIN or to kill a stunned person by shoving a box of crayons up their nostrils.

This is also very "vital" to Clown and Mime gameplay, or so I've convinced myself.


## Changelog
<!-- If necessary, put your changelog entry below. Otherwise, please delete it.
Use however you want to be credited in the changelog in place of CodeDude.
Use (*) for major changes and (+) for minor changes. For example: -->

```changelog
(u)Stonepillar:
(*)You can now put implants in every part of your body, target the desired body part when implanting. To remove them, target the desired body part with scalpel and use the DISARM intent. 
(*)You can now shove crayons into people's noses for minor BRAIN damage with the HARM intent. It will continue to deal BRAIN until it is surgically removed by taking out the implanted crayon from their head.
(+)Micro/macrobombs now have a 60% chance of detonating when not implanted to the chest.
```

zjdtmkhzt commented:
Really, this should add the crayon as an implant, so it can later be surgically extracted.

stonepillars commented:
> 
> 
> Really, this should add the crayon as an implant, so it can later be surgically extracted.

This would lead to the very weird situation of removing the crayon from your chest, but it was shoved up your nose.
A potential future commit, either to this PR or a new one, could add a new surgery route to remove crayons from people's noses but I think it's fine if they stay in there.

Potential to have the medical scanner detect crayons, though.

moonlol commented on code/obj/item/pens_writing_etc.dm:
```diff
@@ -378,6 +378,37 @@
+				return // so you don't accidentally shove a crayon while drawing
+				       // if you want to add this, PLEASE make it obvious
+			else if(check_target_immunity (M))
+				user.show_message( "<span class='alert'>You jam [src] up [M]'s nose, but it wont!</span>")
```
> .... but it wont!
i assume there's meant to be a wont fit?/go in?

> You jam [src] up....
probably could do with a attempt to. "You attempt to jam [src] up..."

stonepillars replied:
Ah, crap. Yeah, that's a typo. This only appears when trying to shove a crayon up the noses of admins, but I'll get on fixing that.

ZeWaka commented on code/obj/item/pens_writing_etc.dm:
```diff
@@ -378,6 +378,37 @@
+			src.set_loc(M)
+			M.take_brain_damage(15) // ouch
+			M.emote("scream")
+			M.reagents.add_reagent("lithium", 3) // drooling for 30s
```
this is a bit counter-intuitive

stonepillars replied:
After discord discussions about it (which start [here](https://discord.com/channels/182249960895545344/298827721604071425/803695963734409217)), it was decided to replace the lithium addition with a SPAWN_DBG or a on_life().

Kyle2143 commented:
I don't think it makes sense for you to be able to put crayons up someone's nose with there being no way for you to remove them.

stonepillars commented:
Addressing the concerns of @Kyle2143 @zjdtmkhzt and @ZeWaka, the lithium has been replaced with a implanted crayon that performs similar effects.

Oh, and the entire implant system was modified to allow you to implant specific body parts.

ZeWaka commented:
> If you somehow managed to put a micro/macrobomb into one of your limbs, it won't work. They need a heartbeat, dummy!
not a good idea imo

changelog doesn't follow format and is way too long, read our code guide

also re: putting everything in commit messages, not a good idea, this isn't git.kernel.org
they will get squashed and removed when merged

instead we do documentation in the code via codedocs so people can actually understand what code is

ZeWaka commented on code/modules/medical/surgery_procs.dm:
```diff
@@ -657,17 +645,44 @@ var/global/list/chestitem_whitelist = list(/obj/item/gnomechompski, /obj/item/gn
+				return 1
+
+/* ---------- SCALPEL - CHEST ---------- */
+	// else if (surgeon.zone_sel.selecting == "chest" && (surgeon.a_intent == "help" || surgeon.a_intent == "disarm"))
```
?

stonepillars replied:
This comment and a lot of the code you requested a change for isn't something I wrote, it's me reordering and moving around the old surgery code. I'll refactor it so it's clean, but I don't actually have an answer for your confusion.

ZeWaka commented on code/modules/medical/surgery_procs.dm:
```diff
@@ -657,17 +645,44 @@ var/global/list/chestitem_whitelist = list(/obj/item/gnomechompski, /obj/item/gn
+/* ---------- SCALPEL - CHEST ---------- */
+	// else if (surgeon.zone_sel.selecting == "chest" && (surgeon.a_intent == "help" || surgeon.a_intent == "disarm"))
+	else if (surgeon.zone_sel.selecting == "chest") // old and new removal works!
+		if (patient.ailments.len > 0)
```
Length()

ZeWaka commented on code/modules/medical/surgery_procs.dm:
```diff
@@ -657,17 +645,44 @@ var/global/list/chestitem_whitelist = list(/obj/item/gnomechompski, /obj/item/gn
+	// else if (surgeon.zone_sel.selecting == "chest" && (surgeon.a_intent == "help" || surgeon.a_intent == "disarm"))
+	else if (surgeon.zone_sel.selecting == "chest") // old and new removal works!
+		if (patient.ailments.len > 0)
+			var/attempted_parasite_removal = 0
```
FALSE

ZeWaka commented on code/modules/medical/surgery_procs.dm:
```diff
@@ -657,17 +645,44 @@ var/global/list/chestitem_whitelist = list(/obj/item/gnomechompski, /obj/item/gn
+			var/attempted_parasite_removal = 0
+			for (var/datum/ailment_data/an_ailment in patient.ailments)
+				if (an_ailment.cure == "Surgery")
+					attempted_parasite_removal = 1
```
TRUE

ZeWaka commented on code/modules/medical/surgery_procs.dm:
```diff
@@ -657,17 +645,44 @@ var/global/list/chestitem_whitelist = list(/obj/item/gnomechompski, /obj/item/gn
+			for (var/datum/ailment_data/an_ailment in patient.ailments)
+				if (an_ailment.cure == "Surgery")
+					attempted_parasite_removal = 1
+					var/success = an_ailment.surgery(surgeon, patient)
+					if (success)
```
inline?

ZeWaka commented on code/modules/medical/surgery_procs.dm:
```diff
@@ -657,17 +645,44 @@ var/global/list/chestitem_whitelist = list(/obj/item/gnomechompski, /obj/item/gn
+					else
+						break
 
+			if (attempted_parasite_removal == 1)
```
No == need

ZeWaka commented on code/modules/medical/surgery_procs.dm:
```diff
@@ -657,17 +645,44 @@ var/global/list/chestitem_whitelist = list(/obj/item/gnomechompski, /obj/item/gn
+				patient, "<span class='alert'>[patient == surgeon ? "You cut" : "<b>[surgeon]</b> cuts"] out a parasite from you with [src]!</span>")
 				return 1
 
+		// Implant removal moved!
```
Unnecessary

stonepillars replied:
Removed.

ZeWaka commented on code/obj/item/implant.dm:
```diff
@@ -1099,20 +1155,23 @@ THROWING DARTS
 
 /obj/item/implanter/mindslave
 	icon_state = "implanter1-g"
+	only_in = "head"
```
Not a fan. Noob trap.

stonepillars replied:
This is something I wanted to keep, but I'm not going to hold up _this_ PR just to get it.

ZeWaka commented on code/modules/medical/surgery_procs.dm:
```diff
@@ -657,17 +645,40 @@ var/global/list/chestitem_whitelist = list(/obj/item/gnomechompski, /obj/item/gn
 
+/* ---------- SCALPEL - CHEST ---------- */
+	else if (surgeon.zone_sel.selecting == "chest") // old and new removal works!
+		if (length(patient.ailments) > 0)
```
```suggestion
		if (length(patient.ailments))
```

ZeWaka commented on code/obj/item/implant.dm:
```diff
@@ -1099,20 +1152,23 @@ THROWING DARTS
 
 /obj/item/implanter/mindslave
 	icon_state = "implanter1-g"
+	only_in = "chest"
```
Still a noob trap. Why is the variable set at all for all this stuff?

stonepillars replied:
For the mindslaver implant, I guess it would be fine to put them in any limb. `only_in` is mainly for microbombs, to prevent people from cutting their own legs off to detonate the mindslave implant. I suppose it could go, now that microbombs just never go off inside of limbs.

stonepillars commented:
Hopefully, this is fine. Microbombs now explode when the limbs they're in are severed. Horrah.

UrsulaMejor commented:
implant anywhere kinda cool. Do medical scanners tell you where the implants are? They should. Foreign obj detected in Right Leg, etc.

not a fan at all of the crayon thing.

UrsulaMejor commented:
1) implanting everywhere creates more busywork for doctors and will lead to a lot more accidental surgery steps
2) shoving crayons up people's nose is incredibly viscerally violent in a way we generally avoid

stonepillars commented:
I'm just going to assume this is being denied. Hooray. I won't delete the branch if you want to steal any of the commits later down the road.
